File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    var answer = 0;
    const sortArray = people.sort((a, b) => a - b);
    while (sortArray.length !== 0) {
        let sum = 0;
        sum += sortArray.pop();
        let temp = true;
        while (temp) {
            if (sortArray[0] + sum > limit || sortArray.length === 0) {
                answer++;
                temp = false;
            }else{
                sum += sortArray.shift();
                answer++;
                temp = false;
            }
        }
    }
    return answer;
}

File: 프로그래머스/2/42885. 구명보트/구명보트.py
from collections import deque

def solution(people, limit):
    answer = 0
    people.sort(reverse=True)
    people = deque(people)

    while people:
        total = people.popleft()
        if people and total + people[-1] <= limit:
            total += people.pop()
        answer += 1

    return answer